pkg/kube/krt/recomputetrigger.go

```diff
@@ -50,3 +50,10 @@ func (r *RecomputeTrigger) TriggerRecomputation() {
 func (r *RecomputeTrigger) MarkDependant(ctx HandlerContext) {
 	_ = Fetch(ctx, r.inner.AsCollection())
 }
+
+// MarkSynced marks this trigger as ready. Before this is called, dependant collections will be blocked.
+// This ensures initial state is populated.
+func (r *RecomputeTrigger) MarkSynced() *RecomputeTrigger {
```

Comment:
```suggestion
func (r *RecomputeTrigger) MarkSynced() {
```

(to echo @hzxuzhonghu - AFAICT this is not needed)